src/util-memcmp.h

```diff
@@ -282,6 +282,94 @@ static inline int SCMemcmpLowercase(const void *s1, const void *s2, size_t len)
     return 0;
 }
 
+#elif defined(__aarch64__)
+#include <arm_neon.h>
+
+#if defined(__ARM_FEATURE_SVE)
+#include <arm_sve.h>
+
+static inline int SCMemcmp(const void *s1, const void *s2, size_t len)
+{
+    uint64_t i = 0;
+    int vec_size = svcntb();
+    svbool_t pg = svwhilelt_b8(i, len);
+    do {
+        svuint8_t b1 = svld1(pg, (uint8_t *)s1 + i);
+        svuint8_t b2 = svld1(pg, (uint8_t *)s2 + i);
+        svbool_t match = svcmpne(pg, b1, b2);
+        if (svptest_any(pg, match))
+            return 1;
+
+        i += vec_size;
+        pg = svwhilelt_b8(i, len);
+    } while (svptest_any(svptrue_b64(), pg));
+    return 0;
+}
+
+#else
+
+/* wrapper around memcmp to match the retvals of the SIMD implementations */
+#define SCMemcmp(a, b, c) ({ memcmp((a), (b), (c)) ? 1 : 0; })
+
+#endif
+
+#define SCMEMCMP_BYTES 16
+
+#define UPPER_LOW   0x41 /* "A" */
+#define UPPER_HIGH  0x5A /* "Z" */
+#define UPPER_DELTA 0x20
+
+static inline int SCMemcmpLowercase(const void *s1, const void *s2, size_t len)
```

Comment:
it looks like this only uses NEON instructions, so it wouldn't see SVE?

Reply:
the other branch of the `#if` uses SVE, this is pure NEON.